src/app/InteractionModelEngine.h

```diff
@@ -429,6 +429,11 @@ class InteractionModelEngine : public Messaging::UnsolicitedMessageHandler,
     // Returns the old data model provider value.
     DataModel::Provider * SetDataModelProvider(DataModel::Provider * model);
 
+#if CHIP_CONFIG_ENABLE_ICD_SERVER && CHIP_CONFIG_ENABLE_ICD_CIP
+    void ScheduledForceTriggerCheckInMessages();
+    void TriggerCheckInMessages(bool aForceSend);
+#endif // CHIP_CONFIG_ENABLE_ICD_CIP && CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
I don't understand why we are adding APIs relevant to the Check-In message in the InteractionModel when the Check-In message has nothing to do with the interaction model.

This creates layer crossing that we should avoid.